cache.js

```diff
@@ -0,0 +1,66 @@
+const MAX_CACHE_SIZE = 100
+
+const generateDefKey = (data, to) => {
+  if (
+    !data ||
+    data.length === 0 ||
+    (!isArrayBuffer(data) && typeof data !== "string")
+  ) {
+    return;
+  }
+
+  const toStr = !!to ? Buffer.from(to).toString("hex") : "";
+
+  let dataStr = data.toString("hex");
+
+  // Trim leading `0x` if present
+  if (dataStr.slice(0, 2) === "0x") {
+    dataStr = dataStr.slice(2);
+  }
+
+  return `${toStr}${dataStr.slice(0, 8)}`.replaceAll(",", "");
```

Comment:
The `replaceAll` is predicated on a literal string of a u8array right? We should not be using that in the key in the first place, so you should remove this call